rts/motoko-rts/src/types.rs

```diff
@@ -540,31 +528,35 @@ impl Concat {
     }
 }
 
-#[repr(packed)]
 pub struct Null {
     pub header: Obj,
 }
 
-#[repr(packed)]
 pub struct Bits64 {
     pub header: Obj,
-    pub bits: u64,
```

Comment:
So was the RTS code actually broken before? Or did the repr(packed) attribute save us?

Reply:
It had undefined behavior because alignment of this field was not as expected by the compiler. We don't know any bugs that this caused so far.

`repr(packed)` is to avoid padding between fields, so it potentially breaks alignment. For this object we don't want any padding between `header` and the payload so we either need `repr(packed)` or to split it into two `u32` fields. The former causes undefined behavior because `u64` fields need to be aligned on 8 byte boundaries (as reported by `core::mem::align_of::<u64>()`), so we now use two `u32` fields.

Reply:
Ok, thanks for the explanation!

Reply:
> We don't know any bugs that this caused so far.

I've added some more assertions in the last commits. It turns out all objects are already aligned as expected so we don't have any bugs.